backend/app/services/git_skill_service.py

```
@@ -0,0 +1,1046 @@
+                if response.status_code != 200:
+                    raise HTTPException(
+                        status_code=response.status_code,
+                        detail=f"Failed to get repository info: {response.text}",
+                    )
+                data = response.json()
+                return data.get("default_branch", "main")
+        except httpx.RequestError as e:
+            raise HTTPException(
+                status_code=503,
+                detail=f"Failed to connect to GitHub API: {str(e)}",
+            )
```

Ruff (0.14.14)

Within an `except` clause, raise exceptions with `raise ... from err` or `raise ... from None` to distinguish them from errors in exception handling

(B904)